### PR TITLE
#1633 Update `pinniped version` output

### PR DESCRIPTION
**Pull request by joshuatcasey**
See #1633 for details.

Note that this does require #1630 to be merged first.

### CODE REVIEW

File: internal/pversion/version.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				info.BuildDate = buildSetting.Value
+			case "vcs.modified":
+				if buildSetting.Value == "false" {
+					info.GitTreeState = "clean"
+				}
+			}
+		}
+	}
+
+	if info.GitVersion == "v0.0.0" && info.GitCommit != "" {
```
Comment (benjaminapetersen) [inline internal/pversion/version.go:76]: Should we make this a named function call?
IE:
```golang
ensureFakeVersionIfNoVersion(foo,bar)
```

File: oidc-auth0.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
```
Comment (benjaminapetersen) [inline oidc-auth0.yaml:1]: Was this file supposed to be included?  As a top level file?

File: Dockerfile [hunk 3]
```diff
@@ -9,6 +9,9 @@ WORKDIR /work
 COPY . .
 ARG GOPROXY
 
+ARG KUBE_GIT_VERSION
```
Comment (benjaminapetersen) [inline Dockerfile:12]: Nice build args, this makes life easier.

**Review by benjaminapetersen — APPROVED** [review]
Nice little polish.

File: internal/pversion/version.go (hunk 1 above)
Comment (benjaminapetersen) [reply internal/pversion/version.go:76]: `ensureFakeDevVersion()` ?

Comment (joshuatcasey) [reply internal/pversion/version.go]: done

**Comment by cfryanr** [comment]
For release notes, and for notifying downstream teams...

This PR changes how the Dockerfile injects the version information into the binary. Anyone who is doing their own `docker build`, or using their own custom Dockerfile, or building the CLI, will need to be aware that they will need to change how they inject the version number, if they choose to inject a version number, during builds.

**Pull request merged**